### DISCUSSION TITLE
Tolgee translation enhancements

### DISCUSSION THREAD

**Discussion by Stephan-P** [Devolutions/UniGetUI#582] [General]
- Automatic (pre-) translation is not available (for Dutch language). Please enable this feature to ease the translators' process.
- If an untranslated text is similar to another text that is already translated, then Tolgee should offer this as a suggestion
- I get the impression that any translator is allowed to declare a translation as Reviewed. Also, any translator appears to be able to change any translation at will, be it a new translation or one that has already been reviewed. In order to prevent chaos and anarchy, we need some hierarchy and levels of edit rights. 
- Tolgee is lacking the following necessary features: e.g. chat, glossary, notifications, easy filtering on 'new' / 'altered' / 'has comment' / 'not translated' / 'not reviewed'

**Comment by Stephan-P** [comment]
- When I open Tolgee - WingetUI - Dutch, it opens Dutch AND English language. The latter I cannot edit. But having these two languages opened severely hampers Tolgee's filter functionality. This default where Dutch AND English languages are opened cannot be changed by me. Annoyance!

**Comment by ppvnf** [comment]
How about something like Crowdin?

Ex: https://crowdin.com/project/playnite

It has many of the features OP mentioned.

**Comment by ppvnf** [comment]
> * When I open Tolgee - WingetUI - Dutch, it opens Dutch AND English language. The latter I cannot edit. But having these two languages opened severely hampers Tolgee's filter functionality. This default where Dutch AND English langueages are opened cannot be changed by me. Annoyance!

about that, you can go to the top-right corner and disable it:
![Captura de tela 2023-01-03 112054](https://user-images.githubusercontent.com/73800734/210375890-3dc2fdce-dfbf-4f44-9422-754e12591cca.png)
![Captura de tela 2023-01-03 112129](https://user-images.githubusercontent.com/73800734/210375893-d3882bf7-587e-406e-a8fe-83295326016e.png)

This is a feature I like about Tolgee, since it allow you to translate/see different languages easily:
![Captura de tela 2023-01-03 112620](https://user-images.githubusercontent.com/73800734/210376660-a4b34c2f-4175-4a51-bf29-18c8fde83a39.png)

**Comment by Stephan-P** [comment]
> How about something like Crowdin?
> Ex: https://crowdin.com/project/playnite
> It has many of the features OP mentioned.

I am familiar with Crowdin and Weblate. And indeed, I was quite suprised to see that WingetUI has opted for this new environment that obviously is not as mature. (I didn't want to shake the tree.)

> you can go to the top-right corner and disable it

Yes, I had noticed the option to remove the English language and I have created my own bookmark. 
I was hoping that the default including English could be adjusted.

**Comment by Stephan-P** [comment]
Oooh, I forgot to mention Transifex, another well established translation environment.

Warning: Ego boosting ahead ;-)

My contributions on Crowdin:
- paint.net
- VidCoder
- Bitwarden (previously)

On Transifex:
- Handbrake
- KeepassXC
- Scribus

On Weblate:
- Gramps
- KeePassDX
- Snipaste
- Vivaldi

**Comment by marticliment** [comment]
Hi @Stephan-P,

the only reason I chose Tolgee was for its pricing. Tolgee is an open-source project, and i'm sure they'll want to hear you out on https://github.com/tolgee/tolgee-platform. It hasn't given me any issues since I started localising.

However, Crowdin looks promising, i'll check it out

**Comment by marticliment** [comment]
I'll move this to a discussion, since i feel like this is not an issue/feature request

**Comment by Stephan-P** [comment]
Accidentally I noticed that there are new strings to translate on Tolgee.
Tolgee does not (yet) have a mechanism by which translators would receive notifications of new strings.
How do you intend to notify the translators of new work to be done for an upcoming release?

**Reply by ppvnf** [reply]
It would be nice to receive an alert when new strings were added, but I think Tolgee should implement this feature

**Reply by marticliment** [reply]
I opened an issue on their repo regarding this, but for the moment theres no ETA

**Reply by Stephan-P** [reply]
We definitely need a mechanism where translators are informed of new strings to be translated, either from Tolgee (long standing request) or via a group message within GitHub.

**Reply by marticliment** [reply]
I will post an announcement each time a big amount of strings is available

**Reply by Stephan-P** [reply]
In which discussion thread would that be, so all translators can subscribe to notifications?

**Reply by marticliment** [reply]
I will create it and let you know once I finish uploading all the new strings
